modules/core/src/main/java/org/apache/ignite/internal/pagemem/wal/record/ConsistentCutFinishRecord.java

```diff
@@ -0,0 +1,105 @@
+import org.apache.ignite.internal.processors.cache.consistentcut.ConsistentCutManager;
+import org.apache.ignite.internal.processors.cache.version.GridCacheVersion;
+import org.apache.ignite.lang.IgniteUuid;
+
+/**
+ * {@link ConsistentCut} splits timeline on 2 global areas - BEFORE and AFTER. It guarantees that every transaction committed
+ * BEFORE also will be committed BEFORE on every other node. It means that an Ignite node can safely recover itself to this
+ * point without any coordination with other nodes.
+ * <p>
+ * This record is written to WAL after Consistent Cut stopped analyzing transactions from {@link ConsistentCut} and
+ * storing them in a particular collection - {@link #before()} or {@link #after()}.
+ * <p>
+ * It guarantees that the BEFORE side consists of:
+ * 1. transactions physically committed before {@link ConsistentCutStartRecord} and weren't included into {@link #after()};
+ * 2. transactions physically committed between {@link ConsistentCutStartRecord} and {@link ConsistentCutFinishRecord}
+ *    and were included into {@link #before()}.
+ * <p>
+ * It guarantees that the AFTER side consists of:
+ * 1. transactions physically committed before {@link ConsistentCutStartRecord} and were included into {@link #after()};
+ * 2. transactions physically committed between {@link ConsistentCutStartRecord} and {@link ConsistentCutFinishRecord}
+ *    and weren't included into {@link #before()}.
+ * <p>
+ * @see ConsistentCutManager
+ */
+public class ConsistentCutFinishRecord extends WALRecord {
+    /** ID of {@link ConsistentCut}. */
+    private final UUID cutId;
+
+    /**
+     * Set of transactions committed between {@link ConsistentCutStartRecord} and {@link ConsistentCutFinishRecord}
+     * to include to the BEFORE side of Consistent Cut.
+     */
+    private final Set<GridCacheVersion> before;
+
+    /**
+     * Set of transactions committed before {@link ConsistentCutStartRecord} to include to the AFTER side of Consistent Cut.
+     */
+    private final Set<GridCacheVersion> after;
+
+    /** */
+    public ConsistentCutFinishRecord(UUID cutId, Set<GridCacheVersion> before, Set<GridCacheVersion> after) {
+        this.cutId = cutId;
+        this.before = Collections.unmodifiableSet(before);
+        this.after = Collections.unmodifiableSet(after);
+    }
+
+    /** */
+    public Set<GridCacheVersion> before() {
+        return before;
+    }
+
+    /** */
+    public Set<GridCacheVersion> after() {
+        return after;
+    }
+
+    /** */
+    public UUID cutId() {
+        return cutId;
+    }
+
+    /** {@inheritDoc} */
+    @Override public RecordType type() {
+        return RecordType.CONSISTENT_CUT_FINISH_RECORD;
+    }
+
+    /** {@inheritDoc} */
+    @Override public String toString() {
+        // Dump IgniteUuid as it more convenient for debug purposes than GridCacheVersion.
+        List<IgniteUuid> txBefore = before.stream()
```

Comment:
Why we need to map sets to `IgniteUuid`? Can, you, provide examples how this will simplify debug or operations?

Reply:
You can check `log.debug` in `ConsistentCut`, `ConsistentCutManager` and `LogCommunicationSpi` in tests. Those records help to track a specific transaction to understand why does it belong to `before` or `after`. It's much simpler to grep log records by single uuid than by GridCacheVersion.

Reply:
Can you, please, provide an example?